### ISSUE TITLE
Use access token to call the HAT API

### ISSUE THREAD

**Issue by thiloplanz** [Hub-of-all-Things/HyperDataBrowser#1]
Currently, you have to give your HAT owner credentials (username/password) to the HyperDataBrowser, which seems like a risky thing to do. Shouldn't this be done by requesting an OAuth access token instead?

Is this possible already on the HAT side of things (i.e. are the necessary API calls for HyperDataBrowser accessible by access token as well)?

**Comment by AndriusA** [comment]
Yes, agree.

This relates to the discussion we had (#15) on authentication mechanisms. Furthermore, the way password is cached in local storage is not a good practice either. The HyperData Browser is a special case however, in that it gives the user full access to their data, and hence has access to all apis, most of which other apps (except for other approved hyperdata browsers) would not get access to.

And yes, many apis are accessible via access token, however we do need to improve authorisation of the remaining ones to allow that. That is a small change and will be done very soon actually.

Regarding oauth, it is a reasonable choice, however it seems to be that it centralises authentication too much and for example for apps it would necessitate each app to sign up for each hat individually the way they do e.g. for facebook, adding a big burden? I'd be very happy to discuss that further however, perhaps in our just launched [forum](http://forum.hubofallthings.com)? Quite empty at the moment, but we're working on making that a go-to resource for any questions about the hat

**Comment by thiloplanz** [comment]
Good point about the problem with OAuth and the requirement to register the app in advance with every single HAT before anyone can use it with that HAT. Since the individual HAT (unlike Facebook) has only a single user, that does seem like an unnecessary step ("service provider" and "end-user" are the same person here).

So access token acquistion would not need to be similar to OAuth. For that part I will look at the forum. Maybe someone has a good idea. 

But having an access token makes sense even for "super-user" apps like HyperData Browser, I believe. Gives you things like possibly very secure authentication during token acquistion (such as 2FA), remote deactivation, scheduled token renewal, per-app access auditing etc.
